### ISSUE TITLE
Add ability to set node flags in tril

### ISSUE THREAD

**Issue by dchopra001** [eclipse/omr#1947]
It would be nice if we had the ability to set flags on a node when create trees in tril. For example, in order to get the optimizer to set the `highWordZero` flag on the `i2l` node, we need to generate the following tree. Note that `Xjit:disableTreeSimplification` needs to be specified so that the tree doesn't fold entirely:

```"(method return=Int64 args=[Int32]  "
" (block                            "
"  (lreturn                         "
"  (land                            "
"      (i2l                         " //basic block extension will realize that this i2l will have highWordZero
"         (iand                     "
"            (iload parm=0)         "
"            (iconst 0x4321)        "
"         )                         "
"      )                            "
"      (lconst %s)))))              ",```

We could reduce this to the following tree if we had the ability to set flags:
```"(method return=Int64 args=[Int32]  "
" (block                            "
"  (lreturn                         "
"  (land                            "
"      (i2l (iload parm=0))         "  //flag would be set here
"      (lconst %s)))))              ",```

**Comment by mgaudet** [comment]
so as a proposed syntax, how does 

    (i2l flags=[HighWordZero,...])

Seem? I suspect this would be comparatively easy to add for a selected subset of the flags, until we had a story for data-driven flags creation.

**Comment by dchopra001** [comment]
I like that syntax. Simple and intuitive.

**Comment by Leonardo2718** [comment]
Although that syntax is very convenient for humans to write, it turns that it can cause some problems for generating Tril from Trees. This is because there are cases where flags are mapped to the same underlying value.

I ran into this problem in my prototype for symbol reference table support in #1552.

Not sure what the extent of this will be for node flags specifically, but it's something to keep in mind.

**Comment by mgaudet** [comment]
In my head, `flags=[FlagA,FlagB,FlagC]` would be implemented as setting `FlagA | FlagB | FlagC`. If FlagA and FlagC share bits, so long as the implementation is faithful, I'm not sure there's an issue; for the Tril Dumper, I definitely see the issue, but not sure I follow for humans writing a test case.

**Comment by Leonardo2718** [comment]
For a human writing Tril, it's not a problem. That's why my symbol reference table prototype allows that syntax, in addition to a couple of others 🙂 

My point was more that the syntax suggested is not enough by itself. There needs to be some additional mechanism that allows the Tril dumper to work and that is not mutually exclusive with the human writable one.

**Comment by mgaudet** [comment]
Right. I just don't think we should block this, at least for a hand-curated subset of flags, on table generation (which, I really should write up an epic for)

**Comment by Leonardo2718** [comment]
> I just don't think we should block this

Fair enough. Not suggesting that we do. Table generation is a separate issue.
